src/sentry/static/sentry/app/components/sidebar/sidebarItem.tsx

```diff
@@ -86,15 +86,16 @@ const SidebarItem = ({
   ...props
 }: Props) => {
   // label might be wrapped in a guideAnchor
+  let labelString = label;
   if (React.isValidElement(label)) {
-    label = label?.props?.children ?? label;
+    labelString = label?.props?.children ?? label;
```

Comment:
I introduced a bug here that only impacted demo mode (the anchors on the sidebar didn't actually mount because we only rendered the child).